### PR TITLE
Constify strings in pref.h

### PR DESCRIPTION
**Pull request by bstoeger**
Make all char * pointers in pref.h const to make it clear that these
strings are not mutable. This meant adding a number of (void *) casts
in calls to free(). Apart from being the right thing to do, this commit
makes the code more consistent, as many of the strings in pref.h were
already const.

Signed-off-by: Berthold Stoeger <bstoeger@mail.tuwien.ac.at>

### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

This patch makes the strings in pref.h consistently const char *. It is the right thing to do and makes the code more consistent.

### CODE REVIEW

File: core/dive.c [hunk 1]
```diff
@@ -3623,11 +3623,12 @@ timestamp_t get_times()
 
 void set_userid(char *rUserId)
 {
+	char *userid = strdup(rUserId);
+	if (strlen(userid) > 30)
+		userid[30]='\0';
```
Comment (neolit123) [inline core/dive.c]: please change the white space too.
`userid[30] = '\0';`
^ space around `=`.

File: core/dive.c [hunk 2]
```diff
@@ -3623,11 +3623,12 @@ timestamp_t get_times()
...
 	if (prefs.userid)
-		free(prefs.userid);
-	prefs.userid = strdup(rUserId);
-	if (strlen(prefs.userid) > 30)
-		prefs.userid[30]='\0';
+		free((void*)prefs.userid);
```
Comment (neolit123) [inline core/dive.c]: please, use `free((void *)....);` here and bellow.
note the space after `void`.

File: core/pref.h [hunk 3]
```diff
@@ -22,9 +22,9 @@ typedef struct
 } partial_pressure_graphs_t;
 
 typedef struct {
-	char *access_token;
-	char *user_id;
-	char *album_id;
+	const char *access_token;
```
Comment (neolit123) [inline core/pref.h:25]: have you enabled `-Wall` for your toolchain to see if some of these changes introduce cast warnings?

**Review by neolit123 — CHANGES_REQUESTED** [review]
thanks for the PR.
added a couple of small comments.

File: core/pref.h (hunk 3 above)
Comment (bstoeger) [reply core/pref.h:25]: Tried `-Wall`. Every single file including "dive.h" gives a signed/unsigned comparison warning. But no new cast warnings.

Comment (neolit123) [reply core/pref.h:25]: yeah, the signed / unsigned i (we) know of. i was hoping that we don't introduce more warnings.

**Comment by neolit123** [comment]
lgtm

**Review by neolit123 — APPROVED** [review]

**Pull request merged**